w-near/test-w-near-defi/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+/*!
+Some hypothetical DeFi contract that will do smart things with the transferred tokens
+*/
+use near_contract_standards::fungible_token::receiver::FungibleTokenReceiver;
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::U128;
+use near_sdk::{
+    env, ext_contract, log, near_bindgen, require, AccountId, Balance, Gas, PanicOnDefault,
+    PromiseOrValue,
+};
+
+const BASE_GAS: u64 = 5_000_000_000_000;
+const PROMISE_CALL: u64 = 5_000_000_000_000;
+const GAS_FOR_FT_ON_TRANSFER: Gas = Gas(BASE_GAS + PROMISE_CALL);
+
+const NO_DEPOSIT: Balance = 0;
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct DeFi {
+    fungible_token_account_id: AccountId,
+}
+
+// Defining cross-contract interface. This allows to create a new promise.
+#[ext_contract(ext_self)]
+pub trait ValueReturnTrait {
+    fn value_please(&self, amount_to_return: String) -> PromiseOrValue<U128>;
+}
+
+// Have to repeat the same trait for our own implementation.
+trait ValueReturnTrait {
+    fn value_please(&self, amount_to_return: String) -> PromiseOrValue<U128>;
+}
+
+#[near_bindgen]
+impl DeFi {
+    #[init]
+    pub fn new(fungible_token_account_id: AccountId) -> Self {
+        require!(!env::state_exists(), "Already initialized");
+        Self {
+            fungible_token_account_id: fungible_token_account_id.into(),
```

Comment:
```suggestion
            fungible_token_account_id,
```

shouldn't be needed, should it? I know it's just used for testing but figured I'd mention